tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,555 @@
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_MUTE,
+        {ATTR_MEDIA_VOLUME_MUTED: True},
+    )
+    # now turn off (stopped and all outputs unselected)
+    get_request_return_values["outputs"] = SAMPLE_OUTPUTS_UNSELECTED
+    await updater_update(["outputs"])
+    await hass.async_block_till_done()
+    # toggle from off
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TOGGLE)
+    # test media play
+    with patch(
+        "homeassistant.components.forked_daapd.media_player.asyncio.sleep",
+        autospec=True,
+    ):
+        task = _service_call(
+            hass,
+            TEST_MASTER_ENTITY_NAME,
+            SERVICE_PLAY_MEDIA,
+            {
+                ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+                ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+            },
+            blocking=False,
+        )  # don't block and gather task later so hass_async_block_till_done() calls inside _update don't get stuck
+        await gather(task)
+        await hass.async_block_till_done()
+        # try playing unsupported media type
+        await _service_call(
+            hass,
+            TEST_MASTER_ENTITY_NAME,
+            SERVICE_PLAY_MEDIA,
+            {
+                ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_TVSHOW,
+                ATTR_MEDIA_CONTENT_ID: "wontwork.mp4",
+            },
+        )
+
+        # throw timeout error in media play
+        def timeout_side_effect(coro, timeout=None):
+            create_task(coro).cancel()
+            raise AsyncioTimeoutError
+
+        with patch("asyncio.wait_for", autospec=True) as wait_for_mock:
+            wait_for_mock.side_effect = timeout_side_effect
+            task = _service_call(
+                hass,
+                TEST_MASTER_ENTITY_NAME,
+                SERVICE_PLAY_MEDIA,
+                {
+                    ATTR_MEDIA_CONTENT_TYPE: MEDIA_TYPE_MUSIC,
+                    ATTR_MEDIA_CONTENT_ID: "somefile.mp3",
+                },
+                blocking=False,
+            )
+            await gather(task)
+            await hass.async_block_till_done()
+
+
+async def test_librespot_java_stuff(
+    hass, mock_api_object, config_entry, get_request_return_values
+):
+    """Test options update and librespot-java stuff."""
+    with patch(
+        "homeassistant.components.forked_daapd.media_player.LibrespotJavaAPI",
+        autospec=True,
+    ):
+        hass.config_entries.async_update_entry(config_entry, options=OPTIONS_DATA)
+        await hass.async_block_till_done()
+    get_request_return_values["player"] = SAMPLE_PLAYER_PLAYING
+    updater_update = mock_api_object.start_websocket_handler.call_args[0][2]
+    await updater_update(["player"])
+    # test media play
+    with patch(
+        "homeassistant.components.forked_daapd.media_player.asyncio.sleep",
```

Comment:
We shouldn't patch sleep. If the code under test uses sleep and we want the tests to be fast, we should patch the time that is slept.